gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -238,9 +240,11 @@ private ManagedChannel createSingleChannel() throws IOException {
 
     ManagedChannelBuilder builder;
 
-    // TODO(weiranf): Add a new API in ComputeEngineCredentials to check whether it's using default
-    // service account.
-    if (isDirectPathEnabled(serviceAddress) && credentials instanceof ComputeEngineCredentials) {
+    // TODO(weiranf): Add API in ComputeEngineCredentials to check default service account.
+    // TODO(weiranf): Remove isDirectPathEnvVarEnabled once setAttemptDirectPath is adapted and the
+    //                env var is removed from client environment.
+    if ((attemptDirectPath || isDirectPathEnvVarEnabled(serviceAddress))
```

Comment:
Not sure if this will over complicate this, but consider tristating attemptDirectPath:
attemptDirectPath is unset  -> fallback to env var
else use attemptDirectPath

Reply:
Done.